src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1630,6 +1654,61 @@ private void MakeSeparatorList(ReadOnlySpan<char> separators, ref ValueListBuild
             }
         }
 
+        private void MakeSeparatorListVectorized(ref ValueListBuilder<int> sepListBuilder, char c, char? c2 = null, char? c3 = null)
+        {
+            Vector256<byte> shuffleConstant = Vector256.Create(0x02, 0x06, 0x0A, 0x0E, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
+                                                        0xFF, 0xFF, 0xFF, 0xFF, 0x02, 0x06, 0x0A, 0x0E, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF);
```

Comment:
I would put the constant in a ROS<byte>, then load it from there. So at runtime it's a load from the assembly's data region, instead of creating the vector on each invocation.

```suggestion
            ReadOnlySpan<byte> shuffleConstantData = new byte[] { 0x02, 0x06, 0x0A, 0x0E, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0x02, 0x06, 0x0A, 0x0E, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF };
            Vector256<byte> shuffleConstant = Unsafe.ReadUnaligned<Vector256<byte>>(ref MemoryMarshal.GetReference(shuffleConstantData));
```

Don't know how accurate [Sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4a9Jq049+ggJI8ZEMfKhKV6zY2btufASyMYovLrmXXqmgMxNSBiEGAG8aBgimfwA1GDAMaFIAViQAHmAATwwYAD4GAFkACgBKUPDIiuIAdgZY+MSU4VhsbMK6BDpSNAZ2uiRu3oBBAY6AURGAMQnJ6Z6EKZnFuYXl2faV9bX5rY3tpc39vbnOkb7T4ePx1cPd2537meKNagqAX3KIj6jauISoZLSmWyeXy5BKZReFUibBg2AAJgB5LiSDKyMTYLjpLK5Bi4AAWHAAZoTpCIPBgMRgACItbAMAC8DC4MAA7gwgTAANoAXVCxy6x36xwuvSuB2uEvFUqO0ruNwekplSt2J0F51OYuVCtl2q1DFezyhkWqDAAqh5sITBDD4ebsJJeABzZlw1J1P4ArHAnKFWCEgowfDQDL5bBQfH2lgAcRgGBhVtgXBUhXxRJJMDJuApPBpFOKTy+72oryAA) is here for `Create`, but when looking at the actual generated asm the difference is similar.

Reply:
Is this some compiler magic to recognize that array as a constant and move it to the assembly's data region? Because under normal circumstances this would allocate, no? I initially thought of moving it to some static field, but this seems like a better solution!

Reply:
Yes, C# compiler optimizes that. It doesn't allocate a new array, instead points to the static data section of the assembly.

See https://vcsjones.dev/2019/02/01/csharp-readonly-span-bytes-static/

Reply:
Ah I see, thanks for the information, had no idea :)